plugins/catalog-backend/src/service/CatalogPlugin.ts

```diff
@@ -164,6 +176,10 @@ export const catalogPlugin = createBackendPlugin({
         });
         builder.addProcessor(...processingExtensions.processors);
         builder.addEntityProvider(...processingExtensions.entityProviders);
+        builder.setEntityDataParser(
+          processingExtensions.entityDataParser ?? defaultEntityDataParser,
+        );
```

Comment:
```suggestion
        if(processingExtensions.entityDataParser){
            builder.setEntityDataParser(
              processingExtensions.entityDataParser,
            );
        }
```
The [`.build()`](https://github.com/backstage/backstage/blob/patch/v1.22.0/plugins/catalog-backend/src/service/CatalogBuilder.ts#L445) method sets the default down below.